networkx/algorithms/operators/product.py

```diff
@@ -532,3 +533,85 @@ def corona_product(G, H):
         GH.add_edges_from((G_node, (G_node, H_node)) for H_node in H)
 
     return GH
+
+
+@not_implemented_for("directed")
+@not_implemented_for("multigraph")
+@nx._dispatchable(graphs=_G_H)
+def modular_product(G, H):
+    r"""Returns the Modular product of G and H.
+
+    The modular product of $G$ and $H$ is the graph $M = G \nabla H$,
+    consisting of the node set $V(M) = V(G) \times V(H)$ that is the Cartesian
+    product of the node sets of $G$ and $H$. Further, $M$ contains an edge $((u, v), (x, y))$:
+
+    - if $u$ is adjacent to $x$ in $G$ and $v$ is adjacent to $y$ in $H$, or
+    - if $u$ is not adjacent to $x$ in $G$ and $v$ is not adjacent to $y$ in $H$.
+
+    More formally, $E(M) = \{((u, v), (x, y)) \mid (u, x) \in E(G) \text{ and } (v, y) \in E(H)\}
+    \cup \{((u, v), (x, y)) \mid (u, x) \notin E(G) \text{ and } (v, y) \notin E(H)\}$
+
+    Parameters
+    ----------
+    G, H: NetworkX graphs
+        The graphs to take the modular product of.
+
+    Returns
+    -------
+    M: NetworkX graph
+        The Modular product of `G` and `H`.
+
+    Raises
+    ------
+    NetworkXNotImplemented
+        If `G` is not a simple graph.
+
+    Examples
+    --------
+    >>> G = nx.cycle_graph(4)
+    >>> H = nx.path_graph(2)
+    >>> M = nx.modular_product(G, H)
+    >>> list(M)
+    [(0, 0), (0, 1), (1, 0), (1, 1), (2, 0), (2, 1), (3, 0), (3, 1)]
+    >>> print(M)
+    Graph with 8 nodes and 8 edges
+
+    References
+    ----------
+    [1] R. Hammack, W. Imrich, and S. Klavžar,
+        "Handbook of Product Graphs", CRC Press, 2011.
+    [2] H. G. Barrow and R. M. Burstall,
+        "Subgraph isomorphism, matching relational structures and maximal
+        cliques", Information Processing Letters, vol. 4, issue 4, pp. 83-84,
+        1976, https://doi.org/10.1016/0020-0190(76)90049-1.
+    [3] V. G. Vizing, "Reduction of the problem of isomorphism and isomorphic
+        entrance to the task of finding the nondensity of a graph." Proc. Third
+        All-Union Conference on Problems of Theoretical Cybernetics. 1974.
+
+    Notes
+    -----
+    The *modular product* is defined in [1] and was first
+    introduced as the *weak modular product*.
```

Comment:
Let's move the "Notes" section above the references and make the references into links within the document.
To do that add double-dot (e.g. `.. [1]`) at the beginning of the reference itself and an underscore where it is referred to (e.g. `[1]_ `.

Also, we don't usually use latex for single letter variables -- and really any variable that can just be typed regularly. That makes it slightly easier to read in the code doc_string, and arguably slightly better in the html doc too.

You can see the html documentation by clicking on "Details" of one of the 32 Checks run automatically. The test is called "documentation artifact". The Details link goes to the home page ofthe documentation and then you navigate to Reference, etc to find the actual page.

None of this is critical, so if you disagree that it looks better or it becomes too much fiddling to too little benefit just say so.

Reply:
Thank you for all the input, I love fiddling :)  Hope it is better now